source/grid/grid_tools.cc

```diff
@@ -429,6 +429,140 @@ namespace GridTools
 
 
 
+  // Generic functions for appending face data in 2D or 3D. TODO: we can
+  // remove these once we have 'if constexpr'.
+  namespace
+  {
+    void
+    append_face_data(const CellData<0> & /*face_data*/,
+                     SubCellData & /*subcell_data*/)
+    {
+      Assert(false, ExcInternalError());
+    }
+
+    void
+    append_face_data(const CellData<1> &face_data, SubCellData &subcell_data)
+    {
+      subcell_data.boundary_lines.push_back(face_data);
+    }
```

Comment:
What do you think of just adding `SubCellData::operator += (const CellData<1> &)` and similarly for `CellData<2>`? Or if not via an operator, do it with a member function to that class?

Reply:
It's fine with me if you want to keep it as is -- it was just an idea.